lib/sqlalchemy/ext/orderinglist.py

```diff
@@ -122,17 +122,23 @@ class Bullet(Base):
 """
 from __future__ import annotations
 
+from operator import index as index_to_int
+from typing import Any
 from typing import Callable
+from typing import Iterable
 from typing import List
 from typing import Optional
+from typing import overload
 from typing import Sequence
+from typing import SupportsIndex
 from typing import TypeVar
+from typing import Union
 
 from ..orm.collections import collection
 from ..orm.collections import collection_adapter
 
 _T = TypeVar("_T")
-OrderingFunc = Callable[[int, Sequence[_T]], int]
+OrderingFunc = Callable[[int, Sequence[_T]], object]
```

Comment:
I'm not what's the latest trend in Any vs object, but here is probably fine to avoid allowing any attribute access like it would happen if using Any

Reply:
Exactly. `Any`, in my view, is a wildcard for the type checker to fill in when it has more detail. `object` is a wildcard for runtime code: we really don't care what this value is, we'll handle it and not touch any of the details.

`Any` is passed on to other code with (where possible), more type detail, `object` is used *here* and it's the most detail on the type the runtime code needs to have.